src/router/index.ts

```diff
@@ -19,6 +26,44 @@ import YearView from "@/features/awards/views/YearView.vue";
 import ProfileView from "@/features/profile/views/ProfileView.vue";
 import { useAuthStore } from "@/stores/auth";
 
+const checkClubAccess = async (
+  to: RouteLocationNormalized,
+  from: RouteLocationNormalized,
+  next: NavigationGuardNext,
+) => {
+  const auth = useAuthStore();
+
+  // Wait for auth initialization to complete
+  while (auth.authLoading) {
+    console.log("waiting for auth to load");
+    await new Promise((resolve) => setTimeout(resolve, 100));
+  }
+
+  // If not logged in, redirect to clubs
+  if (!auth.isLoggedIn) {
+    console.log("redirecting to clubs");
+    return next({ name: "Clubs" });
+  }
+
+  const clubId = to.params.clubId;
+  try {
+    const response = await auth.request.get<ClubPreview[]>("/api/member/clubs");
```

Comment:
Does this request belowing in the auth store or somewhere else? Are we making this request at every nav?